File: 250511/모이자/gather.java
import java.util.Scanner;
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] a = new int[n];
        for (int i = 0; i < n; i++) {
            a[i] = sc.nextInt();
        }
        int minDistanceSum = Integer.MAX_VALUE;
        // Please write your code here.
        for (int i = 0; i < n; i++) {
            int selectHouseIdx = i;

            int distanceSum = 0;
            for (int j = 0; j < n; j++) {
                distanceSum += a[j] * Math.abs(j - selectHouseIdx);
            }
            minDistanceSum = Math.min(minDistanceSum, distanceSum);
        }
        System.out.print(minDistanceSum);
    }
}

// 

File: 251022/모이자/gather.py
n = int(input())
A = list(map(int, input().split()))

import sys

# Please write your code here.

min_total_move = sys.maxsize

for i in range(n) :

    total_move = 0
    for j in range(n) :
        if j == i :
            continue

        total_move += abs(i-j) * A[j]

    min_total_move = min(min_total_move, total_move)

print(min_total_move)